lib/iris/__init__.py

```diff
@@ -303,7 +315,301 @@ def _load_collection(uris, constraints=None, callback=None):
+
+        """
+        # Save the current state
+        saved_settings = self.settings()
+
+        # Apply the new options and execute the context
+        try:
+            self.set(settings, **kwargs)
+            yield
+        finally:
+            # Re-establish the former state
+            self.set(saved_settings)
+
+
+#: Object containing file loading options.
+LOAD_POLICY = LoadPolicy("legacy")
+# The unique (singleton) policy object
+# N.B. FOR NOW, our starting point is "legacy" rather than "default"
+# TODO: resolve tests as needed, to pass with "default".
+
+
+def combine_cubes(cubes, options=None, merge_require_unique=False):
+    """Combine cubes as for load, according to "loading policy" options.
+
+    Applies :meth:`~iris.cube.CubeList.merge`/:meth:`~iris.cube.CubeList.concatenate`
+    steps to the given cubes, as determined by the 'settings'.
+
+    Parameters
+    ----------
+    cubes : list of :class:`~iris.cube.Cube`
+        A list of cubes to combine.
+    options : dict or str
+        Settings, as described for :meth:`iris.LOAD_POLICY.set`.
+        Defaults to current :meth:`iris.LOAD_POLICY.settings`.
+    merge_require_unique : bool
+        Value for the 'unique' keyword in any merge operations.
+
+    Returns
+    -------
+    list of :class:`~iris.cube.Cube`
+
+    .. Note::
+        The ``support_multiple_references`` keyword/property has no effect on the
+        :func:`combine_cubes` operation : it only takes effect during a load operation.
+
+    """
+    if not options:
+        options = LOAD_POLICY.settings()
+    while True:
+        n_original_cubes = len(cubes)
+        sequence = options["merge_concat_sequence"]
+
+        if sequence[0] == "c":
+            # concat if it comes first
+            cubes = cubes.concatenate()
+        if "m" in sequence:
+            # merge if requested
+            cubes = cubes.merge(unique=merge_require_unique)
+        if sequence[-1] == "c":
+            # concat if it comes last
+            cubes = cubes.concatenate()
+
+        # Repeat if requested, *and* this step reduced the number of cubes
+        if not options["repeat_until_unchanged"] or len(cubes) >= n_original_cubes:
+            break
+
+    return cubes
+
+
+def _combine_load_cubes(cubes, merge_require_unique=False):
+    # A special version to call combine_cubes while also implementing the
+    # _MULTIREF_DETECTION behaviour
+    options = LOAD_POLICY.settings()
+    if (
+        options["support_multiple_references"]
+        and "c" not in options["merge_concat_sequence"]
+    ):
+        # Add a concatenate to implement the "multiref triggers concatenate" mechanism
+        from iris.fileformats.rules import _MULTIREF_DETECTION
+
+        if _MULTIREF_DETECTION.found_multiple_refs:
+            options["merge_concat_sequence"] += "c"
+
+    return combine_cubes(cubes, options, merge_require_unique=merge_require_unique)
+
+
+def load(uris, constraints=None, callback=None, policy=None):
```

Comment:
The addition here of "policy=None" is a mistake, as it is not implemented here.
I will remove this when we've reviewed the rest.

So, we might actually add this later,
but if so it should probably have the common style now established elsewhere
 \- i.e. `options=None, **kwargs`
And it should be supported by all of `load`/`load_cube`/`load_cubes`.